sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/ServiceEndpointTest.java

```diff
@@ -0,0 +1,75 @@
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.AsyncDocumentClient;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ServiceEndpointTest extends TestSuiteBase {
+
+    @DataProvider
+    private Object[][] serviceEndpointArgProvider() {
+        return new Object[][] {
+            { TestConfigurations.HOST.substring(0,TestConfigurations.HOST.length() - 1) },
+            { TestConfigurations.HOST },
+            { TestConfigurations.HOST.replace("https://", "") },
+            { TestConfigurations.HOST.replace("https://", "").replace("/", "") },
+            { TestConfigurations.HOST + ";AccountKey=" + TestConfigurations.MASTER_KEY },
+            { TestConfigurations.HOST + ";" + TestConfigurations.MASTER_KEY + ";" },
+            { TestConfigurations.HOST + TestConfigurations.MASTER_KEY + ";" },
+            { TestConfigurations.HOST + TestConfigurations.MASTER_KEY },
+            { null },
+            { "" }
```

Comment:
Can you add a comment for each of these for what they look like?

Reply:
Added comments for the possible service endpoints